metagraph/src/common/algorithms.hpp

```diff
@@ -97,6 +97,73 @@ namespace utils {
         return count;
     }
 
+    // Return true if the two sorted ranges share a common element
+    template <class InputIt1, class InputIt2>
+    constexpr bool share_element(InputIt1 first1,
+                                 InputIt1 last1,
+                                 InputIt2 first2,
+                                 InputIt2 last2) {
+        assert(std::is_sorted(first1, last1));
+        assert(std::is_sorted(first2, last2));
```

Comment:
last -> end ?

Reply:
sure, we can do `begin` and `end`. I was just copying the names used on cppreference